brainglobe_utils/IO/image/load.py

```diff
@@ -732,3 +737,76 @@ def get_size_image_from_file_paths(file_path, file_extension="tif"):
 
     image_shape = {"x": x_shape, "y": y_shape, "z": z_shape}
     return image_shape
+
+
+def get_tiff_meta(
+    path: str,
+) -> Tuple[Tuple[int, int], np.dtype]:
+    with tifffile.TiffFile(path) as tfile:
+        nz = len(tfile.pages)
+        if not nz:
+            raise ValueError(f"tiff file {path} has no pages!")
+        first_page = tfile.pages[0]
+
+    return tfile.pages[0].shape, first_page.dtype
+
+
+lazy_imread = delayed(tifffile.imread)  # lazy reader
+
+
+def read_z_stack(path):
+    """
+    Reads z-stack, lazily, if possible.
+
+    If it's a text file or folder with 2D tiff files use dask to read lazily,
+    otherwise it's a single file tiff stack and is read into memory.
+
+    :param path: Filename of text file listing 2D tiffs, folder of 2D tiffs,
+        or single file tiff z-stack.
+    :return: The data as a dask/numpy array.
+    """
+    if path.endswith(".tiff") or path.endswith(".tif"):
+        with tifffile.TiffFile(path) as tiff:
+            if not len(tiff.series):
+                raise ValueError(
+                    f"Attempted to load {path} but couldn't read a z-stack"
+                )
+            if len(tiff.series) != 1:
+                raise ValueError(
+                    f"Attempted to load {path} but found multiple stacks"
+                )
+
+            axes = tiff.series[0].axes.lower()
+            if set(axes) != {"x", "y", "z"} or axes[0] != "z":
```

Comment:
This is probably too strict? We missed this [on initial review](https://github.com/brainglobe/cellfinder/pull/397), but caught this [on a later review of related functionality](https://github.com/brainglobe/brainglobe-utils/pull/67). I expect we want to be more permissive here based [on previous discussions around how clean we expect TIFF metadata to be](https://github.com/brainglobe/brainglobe-utils/pull/67#issuecomment-2068899564). At the very least, we should be consistent between `read_z_stack` and `get_size_image_from_file_paths` functions?

I suggest doing the equivalent of [`ed40e38` (#67)](https://github.com/brainglobe/brainglobe-utils/pull/67/commits/ed40e386f091a28742acb6d41694014aa3a21f0a) for `read_z_stack`.

Reply:
The code added here is just copied and pasted from cellfinder, so should these changes go into a different PR? Or did I miss something?

Reply:
You didn't miss anything.
Sure, maybe it's better to put it in a different PR.
We should get that PR merged before releasing though, otherwise users that don't have `xyz` (in any order) in their tiff metadata will experience a change in behaviour.